conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureHandler.java

```diff
@@ -32,44 +32,59 @@
             // Servers which do not configure DECODE_URL will still work properly except for encoded slash values.
             // When DECODE_URL has not been disabled, the following handler will no-op
-            (endpoint, handler) -> new URLDecodingHandler(handler, "UTF-8"),
+            endpoint -> handler -> new URLDecodingHandler(handler, "UTF-8"),
             // no-cache and web-security handlers add listeners for the response to be committed,
             // they can be executed on the IO thread.
-            (endpoint, handler) -> Methods.GET.equals(endpoint.method())
+            endpoint -> handler -> Methods.GET.equals(endpoint.method())
                     // Only applies to GET methods
                     ? new NoCachingResponseHandler(handler) : handler,
-            (endpoint, handler) -> new WebSecurityHandler(handler),
-            // It is vitally important to never run blocking operations on the initial IO thread otherwise
-            // the server will not process new requests. all handlers executed after BlockingHandler
-            // use the larger task pool which is allowed to block. Any operation which sets thread
-            // state (e.g. SLF4J MDC or Tracer) must execute on the blocking thread otherwise state
-            // will not propagate to the wrapped service.
-            (endpoint, handler) -> new BlockingHandler(handler),
-            // Logging context and trace handler must execute prior to the exception
-            // to provide user and trace information on exceptions.
-            (endpoint, handler) -> new LoggingContextHandler(handler),
-            (endpoint, handler) -> new TracedOperationHandler(
-                    handler, endpoint.method() + " " + endpoint.template()),
-            (endpoint, handler) -> new ConjureExceptionHandler(handler)
+            endpoint -> handler -> new WebSecurityHandler(handler)
+    ).reverse();
+    // It is vitally important to never run blocking operations on the initial IO thread otherwise
+    // the server will not process new requests. all handlers executed after BlockingHandler
+    // use the larger task pool which is allowed to block. Any operation which sets thread
+    // state (e.g. SLF4J MDC or Tracer) must execute on the blocking thread otherwise state
+    // will not propagate to the wrapped service.
+    private static final ImmutableList<EndpointHandlerWrapper> WRAPPERS_AFTER_BLOCKING =
+            ImmutableList.<EndpointHandlerWrapper>of(
+            endpoint -> handler -> new BlockingHandler(handler),
+    // Logging context and trace handler must execute prior to the exception
+    // to provide user and trace information on exceptions.
+            endpoint -> handler -> new LoggingContextHandler(handler),
+            endpoint -> handler -> new TracedOperationHandler(
+            handler, endpoint.method() + " " + endpoint.template()),
+            endpoint -> handler -> new ConjureExceptionHandler(handler)
```

Comment:
It's probably time to move these into the build method to apply custom wrappers while maintaining readability.
That way we can build a single list of wrappers and apply additional configured wrappers between the right ones in one place.

Reply:
done